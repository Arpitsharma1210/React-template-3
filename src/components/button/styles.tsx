import Button from '@mui/material/Button';
import styled from 'styled-components';

export const StyledButton = styled(Button)<{
  fullWidth?: boolean;
  btnType?: string;
}>`
  padding: ${(props) => (props.btnType === 'primary' ? '8px 18px' : '16px 18px ')} !important;
  white-space: nowrap;
`;
