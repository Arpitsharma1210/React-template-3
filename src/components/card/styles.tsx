import { Card, CardContent } from '@mui/material';
import styled, { css } from 'styled-components';
import { respondTo } from '../../theme/style.layout';
import { brand } from '../../theme/style.palette';

export const StyledCard = styled(Card) <{ bordered?: boolean }>`
    margin-top: 100px;
    ${respondTo.smOnly}{
        margin-top: 0px;
        width:100%;
        height: 100vh;
   }
   ${({ bordered }) => bordered && css`
        border-top : 6px solid ${brand.primaryMain}
   `}
`

export const StyledCardContent = styled(CardContent)`
    ${respondTo.smOnly}{
        padding: 80px 20px !important;
   }
`