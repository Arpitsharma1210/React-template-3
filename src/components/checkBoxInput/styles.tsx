import { Checkbox, FormControlLabel } from "@mui/material";
import { styled } from "styled-components";
import { baseFontFamily, fontSize, fontWeight } from "../../theme/style.typography";
import { brand, colors } from "../../theme/style.palette";



export const StyledFormControlLabel = styled(FormControlLabel)`
    margin-left : 0px !important;
    .MuiFormControlLabel-label {
        font-size: ${fontSize.b2};
        font-weight: ${fontWeight.regular};
        font-family: ${baseFontFamily};
        color: ${brand.textColourDark};
    }
`

export const StyledCheckbox = styled(Checkbox)`

`