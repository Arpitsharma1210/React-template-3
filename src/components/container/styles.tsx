import { Container } from '@mui/material';
import styled from 'styled-components';
import { respondTo } from '../../theme/style.layout';

export interface StyledContainerProps {
    centerAlign?: boolean;
}

export const StyledContainer = styled(Container) <StyledContainerProps>`
    align-items : ${({ centerAlign }) => centerAlign ? 'center' : 'baseline'} !important;
    ${respondTo.smOnly}{
        padding: 0 !important;
    }
`