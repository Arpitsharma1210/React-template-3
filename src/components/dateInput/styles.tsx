import { DatePicker } from "@mui/x-date-pickers";
import { styled } from "styled-components";
import { fontSize, fontWeight } from "../../theme/style.typography";
import { brand } from "../../theme/style.palette";


export const StyledDatePicker = styled(DatePicker)<{fullWidth?:boolean; }>`

    .MuiInputBase-input{
        padding: 10px 16px;
        font-size: ${fontSize.b2};
        font-weight: ${fontWeight.regular};
        color: ${brand.inputBorder};
        &::placeholder {
            color: ${brand.placeHolder};
        }
    }
   
`