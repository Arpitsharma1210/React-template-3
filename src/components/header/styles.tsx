import { styled } from 'styled-components';
import { Typography } from '@mui/material';
import { colors } from '../../theme/style.palette';
import { respondTo } from '../../theme/style.layout';

export const StyledContainer = styled.div`
    display: flex;
    align-items:center;
    flex: 1;
    margin: 16px 0px;
    width: 100%;
`

export const StyledLogoContainer = styled.div`
    margin-right: 32px;
    display: flex;
    ${respondTo.smOnly}{
        margin-left: 32px;
    }
`
export const StyledLogo = styled.img`
    width : 34px;

`

export const StyledNavContainer = styled.div`
    display: flex;
    align-items:center;
    flex: 1;
`

export const StyledNavItem = styled.div<{active?:boolean}>`
    margin-right: 24px;
    padding : 8px;
    border-radius: 8px;
    cursor : pointer;
    background-color: ${({active})=> active ? colors.greyLight : 'transparent'}
`

export const StyledNavItemText = styled(Typography)`

`
export const StyledLogoutContainer= styled.div`
    margin-right: 16px;
    cursor : pointer;
`

export const StyledLogoutText = styled(Typography)`

`