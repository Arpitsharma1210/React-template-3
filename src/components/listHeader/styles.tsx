import { css, styled } from 'styled-components';
import { Typography } from '@mui/material';
import { brand, colors } from '../../theme/style.palette';
import { respondTo } from '../../theme/style.layout';
import { fontSize } from '../../theme/style.typography';

export const StyledContainer = styled.div`
    display: flex;
    align-items: center;
    width: 100%;
    margin: 16px 0px;
    ${respondTo.smOnly}{
        flex-direction: column;
        align-items: baseline;
    }
`

export const StyledHeadingContainer = styled.div`
    margin-right: 32px;
    display: flex;
    ${respondTo.smOnly}{
        margin-left: 32px;
        margin-bottom: 8px;
    }
`
export const StyledHeading = styled(Typography)`
  
`

export const StyledActionItemContainer = styled.div`
    display: flex;
    flex: 1;
    align-items: center;
    justify-content: flex-end;
    ${respondTo.smOnly}{
        flex-direction: column;
        align-items: baseline;
    }
`

export const StyledActionItem = styled.div<{lastItem?:boolean}>`
    margin-right: 24px;
    ${({lastItem})=>lastItem&&css`
        margin-right: 0px;
    `}
    ${respondTo.smOnly}{
        margin: 8px 0px;
    }
`

export const SearchCtaContainer = styled.div`
    display: flex;
    align-items: center;
    padding : 8px 16px;
    cursor:pointer;
    ${respondTo.smOnly}{
        padding-left: 0px;
        margin-bottom: 2px;
    }
`
export const StyledCtaText = styled(Typography)`
  
`

export const SearchInputContainer = styled.div`
    display: flex;
    align-items: center;
    border: 1px solid ${colors.greyMedium};
    border-radius: 8px;

`

export const StyledSearchInput = styled.input`
    min-width: 240px;
    padding : 8px 0px;
    border: none;
    outline: none;
    font-size:${fontSize.b2};
    line-height: 24px;
    &::placeholder {
        color:  ${colors.grey};
    }
`

export const StyledSearchClear = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: ${colors.greyLight};
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
    padding: 10px 8px;
`