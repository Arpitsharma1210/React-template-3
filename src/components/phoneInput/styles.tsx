import PhoneInput from 'react-phone-input-2'
import 'react-phone-input-2/lib/bootstrap.css'
import { css, styled } from 'styled-components';
import { baseFontFamily, fontSize, fontWeight } from '../../theme/style.typography';
import { brand, colors } from '../../theme/style.palette';

export const StyledPhoneInput = styled(PhoneInput)<{error?: boolean; fullWidth?:boolean; readOnly?:boolean;}>`

    .form-control {
        width:100%;
        max-width : ${({fullWidth})=>fullWidth ? '100%' : '384px'};
        outline:none;
        border: 1px solid ${brand.inputBg};
        background-color: ${brand.inputBg};
        border-radius:6px;
        padding: 10px 16px 10px 56px;
        font-size: ${fontSize.b2};
        font-weight: ${fontWeight.regular};
        font-family: ${baseFontFamily};
        color: ${brand.inputBorder};
        &:hover {
            border-color: ${brand.inputBorder};
        }
        &:focus {
            border-color: ${brand.inputBorder};
            background-color: ${colors.white};
            color: ${brand.textColourDark};
            box-shadow: none;
        }

        ${({ error }) => error && css`
            background-color: ${brand.errorBg};
            &:hover {
                border-color: ${brand.errorBorder};
            }
            &:focus {
                border-color: ${brand.errorBorder};
            }
        `}
        ${({ readOnly }) => readOnly && css`
            background-color: transparent;
            cursor: pointer;
            &:hover {
                border-color: ${brand.inputBg};
            }
            &:focus {
                border-color: ${brand.inputBg};
            }
        `}
    }
    .selected-flag.open{
        &:before {
            box-shadow: none;
            border : none;
            outline : none;
        }
    }
    .selected-flag {
        &:focus:before{
            box-shadow: none;
            border : none;
            outline : none;
        }
        outline : none !important;
    }
 
   
`