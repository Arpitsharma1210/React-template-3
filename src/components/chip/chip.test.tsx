import React from 'react';
import Chip from './chip';
import ChipGroup from './chipGroup';
import renderer from 'react-test-renderer';
import { StyledChipContainer, StyledChipShowMoreContainer, StyledPopoverChipGroup, StyledPopoverContainer, StyledPopoverSectionContainer, StyledPopoverSectionHeader, StyledPopoverWrapper } from './styles';


describe('Chip Component', () => {
  it('renders chip with text correctly', () => {
    const tree = renderer
      .create(<Chip text="Test Chip" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders chip with icon when iconCtaClick prop is provided', () => {
    const tree = renderer
      .create(<Chip text="Test Chip" iconCtaClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ChipGroup Component', () => {
  it("renders all chips when displayCount prop is not provided", () => {
    const data = [{ header: 'Group  1', data: [{ key: '1', text: 'Chip  1', bgColor: 'red', textColor: 'white' }] }];
    const tree = renderer
      .create(<ChipGroup data={data} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders only specified number of chips when displayCount prop is provided', () => {
    const data = [{ header: 'Group  1', data: [{ key: '1', text: 'Chip  1', bgColor: 'red', textColor: 'white' }] }];
    const tree = renderer
      .create(<ChipGroup data={data} displayCount={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders chip popover when more chips are available', () => {
    const data = [
      { header: 'Group  1', data: [{ key: '1', text: 'Chip  1', bgColor: 'red', textColor: 'white' }] },
      { header: 'Group  2', data: [{ key: '2', text: 'Chip  2', bgColor: 'blue', textColor: 'white' }] }
    ];
    const tree = renderer
      .create(<ChipGroup data={data} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Styled Components', () => {
  it('renders StyledChipContainer correctly', () => {
    const tree = renderer
      .create(<StyledChipContainer bgColor="red" textColor="white" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });  

  it('renders StyledChipShowMoreContainer correctly', () => {
    const tree = renderer
      .create(<StyledChipShowMoreContainer>Show More</StyledChipShowMoreContainer>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders StyledPopoverWrapper correctly', () => {
    const tree = renderer
      .create(<StyledPopoverWrapper>Popover Content</StyledPopoverWrapper>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders StyledPopoverContainer correctly', () => {
    const tree = renderer
      .create(<StyledPopoverContainer>Popover Container</StyledPopoverContainer>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders StyledPopoverSectionContainer correctly', () => {
    const tree = renderer
      .create(<StyledPopoverSectionContainer>Section Content</StyledPopoverSectionContainer>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders StyledPopoverSectionHeader correctly', () => {
    const tree = renderer
      .create(<StyledPopoverSectionHeader>Section Header</StyledPopoverSectionHeader>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders StyledPopoverChipGroup correctly', () => {
    const tree = renderer
      .create(<StyledPopoverChipGroup>Chip Group</StyledPopoverChipGroup>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});