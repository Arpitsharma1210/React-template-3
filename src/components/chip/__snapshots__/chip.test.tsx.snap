// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Component renders chip with icon when iconCtaClick prop is provided 1`] = `
<div
  className="sc-bcHOTY hmHmYg"
>
  <p
    className="MuiTypography-root MuiTypography-body1 sc-gsMHZj hPfbYd css-ahj2mt-MuiTypography-root"
  >
    Test Chip
  </p>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
    data-testid="CloseRoundedIcon"
    focusable="false"
    onClick={[Function]}
    style={
      {
        "cursor": "pointer",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M18.3 5.71a.9959.9959 0 0 0-1.41 0L12 10.59 7.11 5.7a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
    />
  </svg>
</div>
`;

exports[`Chip Component renders chip with text correctly 1`] = `
<div
  className="sc-bcHOTY hmHmYg"
>
  <p
    className="MuiTypography-root MuiTypography-body1 sc-gsMHZj hPfbYd css-ahj2mt-MuiTypography-root"
  >
    Test Chip
  </p>
</div>
`;

exports[`ChipGroup Component renders all chips when displayCount prop is not provided 1`] = `
<div
  className="MuiGrid-root css-popmxn-MuiGrid-root"
>
  <div
    className="sc-dkHyXG jsyMHN"
    onClick={[Function]}
  >
    <div
      className="sc-bcHOTY hmHmYg"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 sc-gsMHZj hPfbYd css-ahj2mt-MuiTypography-root"
      >
        +1
      </p>
    </div>
  </div>
</div>
`;

exports[`ChipGroup Component renders chip popover when more chips are available 1`] = `
<div
  className="MuiGrid-root css-popmxn-MuiGrid-root"
>
  <div
    className="sc-dkHyXG jsyMHN"
    onClick={[Function]}
  >
    <div
      className="sc-bcHOTY hmHmYg"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 sc-gsMHZj hPfbYd css-ahj2mt-MuiTypography-root"
      >
        +2
      </p>
    </div>
  </div>
</div>
`;

exports[`ChipGroup Component renders only specified number of chips when displayCount prop is provided 1`] = `
<div
  className="MuiGrid-root css-popmxn-MuiGrid-root"
>
  <div
    bgColor="red"
    className="sc-bcHOTY gwcDRY"
    textColor="white"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 sc-gsMHZj hPfbYd css-ahj2mt-MuiTypography-root"
    >
      Chip  1
    </p>
  </div>
</div>
`;

exports[`Styled Components renders StyledChipContainer correctly 1`] = `
<div
  bgColor="red"
  className="sc-bcHOTY gwcDRY"
  textColor="white"
/>
`;

exports[`Styled Components renders StyledChipShowMoreContainer correctly 1`] = `
<div
  className="sc-dkHyXG jsyMHN"
>
  Show More
</div>
`;

exports[`Styled Components renders StyledPopoverChipGroup correctly 1`] = `
<div
  className="sc-iBIYEh gxoWvO"
>
  Chip Group
</div>
`;

exports[`Styled Components renders StyledPopoverContainer correctly 1`] = `
<div
  className="sc-eDgayY cIMjta"
>
  Popover Container
</div>
`;

exports[`Styled Components renders StyledPopoverSectionContainer correctly 1`] = `
<div
  className="sc-jTkTEj iSLCYo"
>
  Section Content
</div>
`;

exports[`Styled Components renders StyledPopoverSectionHeader correctly 1`] = `
<span
  className="sc-gLfKCG bJXxJY"
>
  Section Header
</span>
`;

exports[`Styled Components renders StyledPopoverWrapper correctly 1`] = `
<div
  className="sc-hLljAB bYjhBD"
>
  Popover Content
</div>
`;
